mailpoet/assets/js/src/automation/templates/components/template-detail.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  ComponentType,
+  forwardRef,
+  MouseEventHandler,
+  ReactNode,
+  useCallback,
+  useState,
+} from 'react';
+import apiFetch from '@wordpress/api-fetch';
+import { Button, Modal, Snackbar as WpSnackbar } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { chevronLeft, chevronRight, warning } from '@wordpress/icons';
+import { Tag } from '@woocommerce/components';
+import { addQueryArgs } from '@wordpress/url';
+import { TemplatePreview } from './template-preview';
+import { AutomationTemplate, automationTemplateCategories } from '../config';
+import { MailPoet } from '../../../mailpoet';
+
+// snackbar icon is not annotated in the types
+const Snackbar = WpSnackbar as ComponentType<
+  WpSnackbar.Props & {
+    icon: ReactNode;
+  }
+>;
+
+const getCategory = (template: AutomationTemplate): string =>
+  automationTemplateCategories.find(({ slug }) => slug === template.category)
+    ?.name ?? __('Uncategorized', 'mailpoet');
+
+const useCreateFromTemplate = () => {
+  const [state, setState] = useState({
+    data: undefined,
+    loading: false,
+    error: undefined,
+  });
+
+  const create = useCallback(async (slug: string) => {
+    setState((prevState) => ({ ...prevState, loading: true }));
+    try {
+      const data = await apiFetch<{ data: { id: string } }>({
+        path: `/automations/create-from-template`,
+        method: 'POST',
+        data: { slug },
+      });
+      MailPoet.trackEvent('Automations > Template selected', {
+        'Automation slug': slug,
+      });
+      window.location.href = addQueryArgs(MailPoet.urls.automationEditor, {
+        id: data.data.id,
+      });
+    } catch (error) {
+      setState((prevState) => ({ ...prevState, error }));
+    } finally {
+      setState((prevState) => ({ ...prevState, loading: false }));
+    }
+  }, []);
+
+  return [create, state] as const;
+};
+
+type Props = {
+  template: AutomationTemplate;
+  onRequestClose: Modal.Props['onRequestClose'];
+  onPreviousClick?: MouseEventHandler<HTMLButtonElement>;
+  onNextClick?: MouseEventHandler<HTMLButtonElement>;
+};
+
+export const TemplateDetail = forwardRef<HTMLDivElement, Props>(
+  ({ template, onRequestClose, onPreviousClick, onNextClick }, ref) => {
+    const [createAutomationFromTemplate, { loading, error }] =
+      useCreateFromTemplate();
+
+    return (
+      <Modal
+        ref={ref}
+        className="mailpoet-automation-template-detail"
+        title=""
+        onRequestClose={onRequestClose}
+      >
+        <div className="mailpoet-automation-template-detail-content">
+          <div className="mailpoet-automation-template-detail-info">
+            <Tag label={getCategory(template)} />
+            <h1>{template.name}</h1>
+            {template.description}
+          </div>
+          <div className="mailpoet-automation-template-detail-preview">
```

Comment:
Currently, if the user clicks multiple times on the left or right arrows, we are going to trigger a request to load the preview for several automations, but maybe we only need to load the last automation as, ultimately, that is the only one that will be displayed to the user? Also, the description of the automation loads faster than the preview, causing a weird behavior where the browser displays the description of the last automation, but we are still showing the preview of previous automations.

I don't think this is a blocker but more a suggestion for improvement.

Reply:
@rodrigoprimo
> Currently, if the user clicks multiple times on the left or right arrows, we are going to trigger a request to load the preview for several automations, but maybe we only need to load the last automation as, ultimately, that is the only one that will be displayed to the user?

I'm not too worried about a few extra requests in the admin, but this is still a problem because it will cause the arriving templates to "blink", and it may end up on a different one (race condition). Fixed with abort controller: https://github.com/mailpoet/mailpoet/pull/5310/commits/8764064d11ee3e25019889e5e690f0ee0af8ca72

> Also, the description of the automation loads faster than the preview, causing a weird behavior where the browser displays the description of the last automation, but we are still showing the preview of previous automations.

Do you see a preview of previous automations? There should be a spinner:

https://github.com/mailpoet/mailpoet/assets/141436/eccbbdbe-b0d7-4a6d-9e96-9673010a5931

Reply:
Thanks for the fix.